src/main/java/io/snyk/jenkins/SnykStepFlow.java

```diff
@@ -17,53 +17,57 @@
 
 public class SnykStepFlow {
 
-  public static void perform(SnykConfig config, Supplier<SnykContext> contextSupplier) throws SnykIssueException, SnykErrorException {
-    int testExitCode = 0;
+  public static void perform(SnykConfig config, Supplier<SnykContext> contextSupplier)
+  throws SnykIssueException, SnykErrorException {
+    boolean foundIssues = false;
     Exception cause = null;
     SnykContext context = null;
 
     try {
       context = contextSupplier.get();
-      testExitCode = SnykStepFlow.scan(context, config);
+
+      SnykInstallation installation = SnykInstallation.install(
+        context,
+        config.getSnykInstallation()
+      );
+
+      context.getEnvVars().put("SNYK_TOKEN", SnykApiToken.getToken(context, config.getSnykTokenId()));
+
+      foundIssues = SnykStepFlow.testProject(context, config, installation);
```

Comment:
Nit: not sure if "testProject" returning a boolean whether issues were found is the best method name. What about (but that's hiding that it's going to test the project 🤔):

```suggestion
      foundIssues = SnykStepFlow.projectHasIssues(context, config, installation);
```

Reply:
@JackuB There's already a SnykTest.Result, we can return that and let this method pick out `result.foundIssues`?

Reply:
There's already a SnykTest.Result, we can return that and let this method pick out result.foundIssues?

(I repeated this comment so GitHub registers it as a reply)

Reply:
Tried it. Not as simple since the SnykTest.Result.jsonFIle is cleaned up before returning so it's misleading to have it available. I'd need a new Result class (more code) or squash `testProject` into `perform` (which then makes it slightly harder to read). Not a fan of using `projectHasIssues` for the reason you mentioned.

I'm going to merge. We can consider improvements here in future refactorings.